scipy/signal/tests/test_signaltools.py

```diff
@@ -777,6 +777,17 @@ def test_many_sizes(self, n):
         out = fftconvolve(a, b, 'full', axes=[0])
         assert_allclose(out, expected, atol=1e-10)
 
+    def test_fft_nan(self):
+        n = 1000
+        rng = np.random.default_rng(43876432987)
+        sig_nan = rng.standard_normal(n)
+
+        for val in [np.nan, np.inf]:
+            sig_nan[100] = val
+            coeffs = signal.firwin(200, 0.2)
+
+            with pytest.warns(UserWarning):
```

Comment:
```suggestion
            with pytest.warns(UserWarning, match='Use of fft convolution'):
```
Good practice to be more specific, even if the code doesn't emit other `UserWarning`s, as there have been cases where an unintended warning/error of the same time is emitted.